src/cuckoo/mean.hpp

```diff
@@ -1051,8 +1052,9 @@ class solver_ctx {
     showcycle = show_cycle;
     cuckoo = 0;
   }
-  void setheadernonce(char* const headernonce, const u32 len, const u32 nonce) {
-    ((u32 *)headernonce)[len/sizeof(u32)-1] = htole32(nonce); // place nonce at end
+  void setheadernonce(char* const headernonce, const u32 len, const u64 nonce) {
+    // The KeyHash takes 44 byte - put nonce at 45-56
```

Comment:
I see correct according to [related comment](https://github.com/aeternity/epoch/blob/515e38e8ee59683f38cb148f5279934066fb2e76/apps/aecore/src/aec_pow_cuckoo.erl#L336-L352):
> The SHA256 hash is 32 bytes (44 chars base64-encoded), the nonce is 8 bytes (12 chars base64-encoded).

[Doc](https://github.com/aeternity/protocol/blob/9b2c049c0a8878aa266f8bc18a492a7f9ecf6275/consensus/consensus.md#proof-of-work-1) may benefit from improvement.